frontend/src/entities/group/detail/group-detail-user.actions.tsx

```diff
@@ -0,0 +1,121 @@
+import { Button, ButtonDropdown, Icon, SpaceBetween } from '@cloudscape-design/components';
+import { getGroupUsers, removeGroupUser } from '../group.reducer';
+import React, { useState } from 'react';
+import { Action, Dispatch, ThunkDispatch } from '@reduxjs/toolkit';
+import { NavigateFunction, useNavigate, useParams } from 'react-router-dom';
+import { setDeleteModal } from '../../../modules/modal/modal.reducer';
+import { ModalProps } from '../../../modules/modal';
+import { IGroupUser } from '../../../shared/model/groupUser.model';
+import NotificationService from '../../../shared/layout/notification/notification.service';
+import AddGroupUserModal from './add-group-user-modal';
+import { IUser } from '../../../shared/model/user.model';
+
+function GroupDetailUserActions (props?: any) {
+    const dispatch = useAppDispatch();
+    const navigate = useNavigate();
+    const {groupName} = useParams();
+    const allUsers: IUser[] = useAppSelector((state) => state.user.allUsers) || [];
+    const groupUsers: IGroupUser[] = useAppSelector((state) => state.group.currentGroupUsers) || [];
+    const groupUsernames = groupUsers.map((user) => user.user);
+    const addableUsers = allUsers.filter((user) => !groupUsernames.includes(user.username!));
+    const [addUserModalVisible, setAddUserModalVisible] = useState(false);
+
+    return (
+        <SpaceBetween direction='horizontal' size='xs'>
+            <AddGroupUserModal dispatch={dispatch} setVisible={setAddUserModalVisible} visible={addUserModalVisible} addableUsers={addableUsers} groupName={groupName}/>
+            <Button onClick={() => dispatch(getGroupUsers(groupName))} ariaLabel={'Refresh groups list'}>
+                <Icon name='refresh'/>
+            </Button>
+            {GroupDetailUserActionsButton(navigate, dispatch, props)}
+            <Button
+                variant='primary'
+                onClick={() => setAddUserModalVisible(true)}
+            >
+                Add User
+            </Button>
+        </SpaceBetween>
+    );
+}
+
+function GroupDetailUserActionsButton (navigate: NavigateFunction, dispatch: Dispatch, props?: any) {
+    const selectedUser: IGroupUser = props?.selectedItems[0];
+    const items = [];
+    if (selectedUser) {
+        items.push({
+            text: 'Remove from Group',
+            id: 'removeFromGroup',
+        });
+    }
+
+    const [modalState, setModalState] = React.useState<Partial<ModalProps>>({
+        visible: false,
+        confirmText: 'Confirm',
+        dismissText: 'Cancel',
+    });
+
+    return (
+        <ButtonDropdown
+            items={items}
+            variant='primary'
+            disabled={!selectedUser}
+            onItemClick={(e) => GroupDetailUserActionHandler(e, dispatch, modalState as ModalProps, setModalState, selectedUser)}
+        >
+            Actions
+        </ButtonDropdown>
+    );
+}
+
+const GroupDetailUserActionHandler = async (
+    e: any,
```

Comment:
I know this is done other places in the code but `any` should be `CustomEvent<ButtonDropdownProps.ItemClickDetails>`